common/txmgr/strategies.go

```diff
@@ -56,14 +56,14 @@ func (s DropOldestStrategy) Subject() uuid.NullUUID {
 	return uuid.NullUUID{UUID: s.subject, Valid: true}
 }
 
-func (s DropOldestStrategy) PruneQueue(ctx context.Context, pruneService txmgrtypes.UnstartedTxQueuePruner) (n int64, err error) {
+func (s DropOldestStrategy) PruneQueue(ctx context.Context, pruneService txmgrtypes.UnstartedTxQueuePruner) (ids []int64, err error) {
 	var cancel context.CancelFunc
 	ctx, cancel = context.WithTimeout(ctx, s.queryTimeout)
 	defer cancel()
 
-	n, err = pruneService.PruneUnstartedTxQueue(ctx, s.queueSize, s.subject)
+	ids, err = pruneService.PruneUnstartedTxQueue(ctx, s.queueSize, s.subject)
```

Comment:
You might also want to change 2nd param to "s.queueSize - 1", to reduce the queueSize by 1.
This is because previously, it used to prune AFTER the new Tx was added, and now it is pruning BEFORE.

So previously if queueSize was 250, then this limit was enforced after we inserted the new tx.
But now, we first prune, then add new tx. So the pruning logic should prune to 249, not 250.

It is kind of a nit, but stil.